cli/tools/registry/mod.rs

```diff
@@ -635,6 +640,53 @@ async fn publish_package(
     package.package,
     package.version
   );
+
+  if provenance {
+    // Get the version manifest from JSR
+    let meta_url = jsr_url().join(&format!(
+      "@{}/{}/{}_meta.json",
+      package.scope, package.package, package.version
+    ))?;
+
+    let meta_bytes = client.get(meta_url).send().await?.bytes().await?;
+
+    if std::env::var("DISABLE_JSR_MANIFEST_VERIFICATION_FOR_TESTING").is_err() {
```

Comment:
This is fine for a first pass, but we should look into removing this env var so we're properly testing this functionality and so that someone doesn't get this check disabled somehow.

Reply:
Yeah, we need the registry test_server to handle package uploads and manifest generation for uploaded packages.